core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -101,29 +107,161 @@
   private final boolean inHeap;
 
   /**
+   * The size of memory granted to each consumer.
+   */
+  private final HashMap<MemoryConsumer, Long> consumers;
```

Comment:
Rather than maintaining a separate bookkeeping map, why not track the amount of memory granted to each `MemoryConsumer` inside of the consumer itself, then expose a method to let us retrieve that size from the consumer instance?